routes/create-or-update-file.js

```diff
@@ -0,0 +1,65 @@
+var request = require("request");
+
+module.exports = function(config) {
+  return function(req, res) {
+    if(!config.authorized(req, res)) {
```

Comment:
An elegant way to incorporate this change would be for the `module.exports` function to return an array:

```js
module.exports = function(config) {
  return [
    config.authorized,
    function(req, res) { 
      // ...
    }
  ];
};
```